tests/test_activate.py

```diff
@@ -1,652 +1,582 @@
+                assert_in("deactivate must be sourced", stderr)
+                assert_in("Usage: source deactivate", stderr)
 
             commands = (command_setup + """
-            source {deactivate} --help
-            """).format(envs=envs, deactivate=deactivate)
-
+            {source} {syspath}{binpath}deactivate --help
+            """).format(envs=envs, **_format_vars)
             stdout, stderr = run_in(commands, shell)
-            assert stdout == ''
-            assert "Usage: source deactivate" in stderr
-
+            assert_equals(stdout, '')
+            if platform == 'win':
+                assert_in("Usage: deactivate", stderr)
+            else:
+                assert_in("Usage: source deactivate", stderr)
 
 @pytest.mark.slow
 def test_activate_symlinking():
     for shell in shells:
         with TemporaryDirectory(prefix='envs', dir=dirname(__file__)) as envs:
-            activate, deactivate, conda = _write_entry_points(envs)
             commands = (command_setup + """
-            source {activate} {envs}/test1
-            """).format(envs=envs, activate=activate)
-
+            {source} {syspath}{binpath}activate "{env_dirs[0]}"
+            """).format(envs=envs, env_dirs=gen_test_env_paths(envs), **_format_vars)
             stdout, stderr = run_in(commands, shell)
-            assert not stdout
-            assert stderr == 'discarding {syspath} from PATH\nprepending {envs}/test1/bin to PATH\n'.format(envs=envs, syspath=syspath)
-            for f in ['conda', 'activate', 'deactivate']:
-                assert os.path.lexists('{envs}/test1/bin/{f}'.format(envs=envs, f=f))
-                assert os.path.exists('{envs}/test1/bin/{f}'.format(envs=envs, f=f))
-                s = os.lstat('{envs}/test1/bin/{f}'.format(envs=envs, f=f))
-                assert stat.S_ISLNK(s.st_mode)
-                assert os.readlink('{envs}/test1/bin/{f}'.format(envs=envs,
-                    f=f)) == '{syspath}/{f}'.format(syspath=syspath, f=f)
-
-            try:
-                # Test activate when there are no write permissions in the
-                # env. There are two cases:
-                # - conda/deactivate/activate are already symlinked
-                commands = (command_setup + """
-                mkdir -p {envs}/test3/bin
-                ln -s {activate} {envs}/test3/bin/activate
-                ln -s {deactivate} {envs}/test3/bin/deactivate
-                ln -s {conda} {envs}/test3/bin/conda
-                chmod 555 {envs}/test3/bin
-                source {activate} {envs}/test3
-                """).format(envs=envs, activate=activate, deactivate=deactivate, conda=conda)
-                stdout, stderr = run_in(commands, shell)
-                assert not stdout
-                assert stderr == 'discarding {syspath} from PATH\nprepending {envs}/test3/bin to PATH\n'.format(envs=envs, syspath=syspath)
+            assert_equals(stderr, u'prepending {envpaths1} to PATH'\
```

Comment:
Should be
```python
assert_equals(stderr, u'discarding {syspath} from PATH\nprepending {envpaths1} to PATH'\
```
as the root env should first be discarded before the test1 env is being added.